components/director/internal/domain/labeldef/service.go

```diff
@@ -43,3 +46,29 @@ func (s *service) Create(ctx context.Context, def model.LabelDefinition) (model.
 	return def, nil
 
 }
+
+func (s *service) Update(ctx context.Context, def model.LabelDefinition) (model.LabelDefinition, error) {
+	if err := def.ValidateForUpdate(); err != nil {
+		return model.LabelDefinition{}, errors.Wrap(err, "while validating Label Definition")
+	}
+
+	ld, err := s.repo.GetByKey(ctx, def.Tenant, def.Key)
+	if err != nil {
+		return model.LabelDefinition{}, errors.Wrap(err, "while receiving Label Definition")
+	}
+
+	id := ld.ID
+	ld = &def
+	ld.ID = id
+
+	if err := s.repo.Update(ctx, *ld); err != nil {
```

Comment:
what about checking if existing labels are valid against the new definition?

Reply:
done